File: py/basic/1057.py
# N, p, q = map(int, input().split())


# def check(num):
#     if num%2 == 1:
#         num = (num+1)//2
#     else:
#         num = num//2
#     return num 


# n = 0
# while True:
#     p = check(p)
#     q = check(q)

#     if p == q:
#         n+=1
#         print(n)
#         break

#     n+=1



def check(num):
    if num % 2 == 1:
        num = (num + 1) // 2
    else:
        num = num // 2
    return num

def find_n(p, q, n=0):
    if p == q:
        return n
    else:
        p = check(p)
        q = check(q)
        return find_n(p, q, n + 1)

N, p, q = map(int, input().split())
result = find_n(p, q)
print(result)

File: cpp/basic/1057.cpp
//백준 1057
#include <iostream>

using namespace std;

int check(int num);

int main() {
    int N;
    int p;
    int q;
    int n = 0;
    cin >> N >> p >> q;

    while (true) {
        p = check(p);
        q = check(q);
        if (p==q) {
            n = n+1;
            cout << n;
            break;
        }
        n = n+1;
    }
}


int check(int num) {
    if (num % 2 == 1) {
        num = (num+1) / 2;
    } else {
        num = num / 2;
    }

    return num;
}